txinput.go

```diff
@@ -2,14 +2,26 @@ package bt
 
 import (
 	"bytes"
+	"context"
 	"encoding/binary"
+	"encoding/hex"
+	"errors"
 	"fmt"
 
 	"github.com/libsv/go-bk/crypto"
 
 	"github.com/libsv/go-bt/v2/bscript"
 )
 
+// ErrNoUTXO signals the UTXOGetterFunc has reached the end of its input.
+var ErrNoUTXO = errors.New("no remaining utxos")
+
+// UTXOGetterFunc is used for tx.Fund. It expects []*bt.UTXO to be returned containing
+// utxos of which an input can be built.
+//
+// It is expected that bt.ErrNoUTXO will be returned once the utxo source is depleted.
+type UTXOGetterFunc func(ctx context.Context, deficit uint64) ([]*UTXO, error)
```

Comment:
maybe just call deficit 'satoshis' and add in the comment that satoshis is the amount remaining to be funded.

Reply:
Or 'remaining'

Reply:
Deficit does mean remaining though?